Riot/Modules/Room/Views/BubbleCells/RoomTypingBubbleCell.swift

```diff
@@ -0,0 +1,125 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import UIKit
+
+@objcMembers
+class RoomTypingBubbleCell: UITableViewCell {
+    // MARK: - Constants
+    
+    static let cellIdentifier = "RoomTypingBubbleCell"
+    
+    // MARK: - Outlets
+    
+    @IBOutlet weak var additionalUsersLabel: UILabel?
+    @IBOutlet weak var additionalUsersLabelLeadingConstraint: NSLayoutConstraint?
+    @IBOutlet weak var dotsView: DotsView?
+    @IBOutlet weak var dotsViewLeadingConstraint: NSLayoutConstraint?
+
+    // MARK: - members
+    
+    private var userPictureViews: Array<MXKImageView> = Array()
+    
+    // MARK: - Lifecycle
+    
+    override func awakeFromNib() {
+        super.awakeFromNib()
+        
+        additionalUsersLabel?.textColor = ThemeService.shared().theme.textSecondaryColor
+        dotsView?.highlightedDotColor = ThemeService.shared().theme.textTertiaryColor
+        dotsView?.dotColor = ThemeService.shared().theme.textSecondaryColor
+    }
+    
+    override func prepareForReuse() {
+        super.prepareForReuse()
+        
+        for pictureView in userPictureViews {
+            pictureView.removeFromSuperview()
+        }
+        
+        additionalUsersLabel?.textColor = ThemeService.shared().theme.textSecondaryColor
+        dotsView?.highlightedDotColor = ThemeService.shared().theme.textTertiaryColor
+        dotsView?.dotColor = ThemeService.shared().theme.textSecondaryColor
+    }
+    
+    override func layoutSubviews() {
+        super.layoutSubviews()
+
+        self.dotsView?.isHidden = userPictureViews.count == 0
+
+        guard userPictureViews.count > 0 else {
+            return
+        }
+
+        additionalUsersLabel?.sizeToFit()
+        
+        var pictureViewsMaxX: CGFloat = 0
+        var xOffset: CGFloat = 0
+        for pictureView in userPictureViews {
+            pictureView.center = CGPoint(x: 16 + xOffset + pictureView.bounds.midX, y: self.bounds.midY)
+            xOffset += round(pictureView.bounds.maxX * 2 / 3)
+            pictureViewsMaxX = pictureView.frame.maxX
+        }
+        
+        let leftMagin: CGFloat = pictureViewsMaxX + (userPictureViews.count == 1 ? 16 : 8)
+        additionalUsersLabelLeadingConstraint?.constant = leftMagin
+        
+        dotsViewLeadingConstraint?.constant = additionalUsersLabel?.text.isEmptyOrNil == true ? leftMagin : leftMagin + 8 + (additionalUsersLabel?.frame.width ?? 0)
+    }
+    
+    // MARK: - Business methods
+    
+    func updateTypingUsers(_ typingUsers: Array<TypingUserInfo>, mediaManager: MXMediaManager) {
+        for pictureView in userPictureViews {
+            pictureView.removeFromSuperview()
+        }
+        userPictureViews = Array()
+        
+        for user in typingUsers {
+            if userPictureViews.count >= 4 {
```

Comment:
This one deserves to be a constant i think. You could also consider defining constants for other magic numbers in this method.